src/databricks/labs/blueprint/wheels.py

```diff
@@ -232,6 +232,26 @@ def upload_to_wsfs(self) -> str:
             self._installation.save(Version(self._product_info.version(), remote_wheel, self._now_iso()))
             return remote_wheel
 
+    def upload_wheel_dependencies(self, prefixes: list[str] | None = None) -> list[str]:
+        """Uploads the wheel dependencies to WSFS location of installation and returns the remote paths.
+        :param prefixes (optional): A list of prefixes to match against the wheel names. If a prefix matches, the wheel is uploaded. If None, all wheels are uploaded. Defaults to None.
+        """
+        if prefixes is None:
+            prefixes = [wheel.name for wheel in self._local_wheel]
+        remote_paths = []
+        for wheel in self._local_wheel:
+            for prefix in prefixes:
+                # TODO: This function should be modified as there is no guarantee that the main wheel.name contains product name in it
+                # TODO: In the future the need of passing explicit prefixes should be removed
+                if (
+                    prefix in wheel.name
+                    and wheel.name.endswith("-none-any.whl")
+                    and self._product_info.product_name() not in wheel.name
+                ):
```

Comment:
```suggestion
                if not wheel.name.endswith("-none-any.whl"): # move this filter to the upper loop, as it has nothing to do with prefix
                    continue
                if wheel.name == self._local_wheel.name:
                    # main wheel has to be uploaded with upload_to_wsfs() method already.
                    continue
                if not wheel.name.startswith(prefix):
                    continue
```
simplify and make it more readable.